arches/app/search/components/base.py

```diff
@@ -40,7 +41,15 @@ def view_data(self):
 
         pass
 
-    def post_search_hook(self, search_results_object, results, permitted_nodegroups):
+    def execute_query(self, search_results_object, response_object):
```

Comment:
I think this method should be put into a new `BaseSearch` class.
This class can be implemented by `core_search.py`.

Reply:
I've been thinking about this one a lot and I think it still makes sense to have `execute_query` not only be its own method to compartmentalize the logic but also that it originate from the `BaseSearchFilter` class as opposed to the `CoreSearchComponent` class and here's why:  

Search Components that are not Core Search Components should still be able to insert logic at this specific point between `append_dsl` and `post_search_hook`. In the example of [this draft PR](https://github.com/archesproject/arches/pull/10766) where there's a follow-up query to get all those resources related to what's in the result set, it would be ideal for this action to happen *before* `post_search_hook` so that all the logic regarding how to structure or format results applies to this superset of results.